File: lib/services/posts.dart
import 'package:develove/models/post.dart';
import 'package:develove/services/constants.dart';

class PostServices {
  static Future<List<Post>> getPosts() async {
    List<Post> posts = [];
    final res = await supabase.from('posts').select().execute();
    print(res.status);
    for (int i = 0; i < res.data.length; i++) {
      final data = res.data[i];
      posts.add(Post(
          pid: data['pid'],
          uid: data['uid'],
          title: data['title'],
          content: data['data'],
          hearts: data['hearts'],
          tags: (data['tags'] as List<dynamic>)
              .map((e) => e.toString())
              .toList()));
    }
    return posts;
  }
}

printPosts() async {
  print(await PostServices.getPosts());
}

File: web/lib/posts.js
import { supabase } from "./supabase";

export async function createPosts(posts) {
  let currentUser = supabase.auth.user().email;
  let userRes = await supabase
    .from("users")
    .select("uid")
    .match("email", currentUser);
  let data = {
    uid: userRes.data[0].uid,
    title: posts.title,
    hearts: 0,
    data: posts.data,
    tags: posts.tags,
  };
  await supabase.from("posts").insert(data);
}

export async function getPosts() {
  let currentUser = supabase.auth.user().email;
  let userRes = await supabase
    .from("users")
    .select("uid")
    .match("email", currentUser);
  let posts = await supabase
    .from("posts")
    .select()
    .match("uid", userRes.data[0].uid);
  return posts.data;
}
